engines/engine.go

```diff
@@ -44,6 +40,10 @@ type Engine interface {
 	// from this method.
 	PayloadSchema() runtime.CompositeSchema
 
+	// ConfigSchema returns the CompositeSchema that represents the engine
+	// configuration
+	ConfigSchema() runtime.CompositeSchema
```

Comment:
Wrong place, this should be added to the `EngineProvider` which should be changed from function to interface.